packages/framework/src/Model/Product/Search/Export/ProductSearchExportScheduler.php

```diff
@@ -4,36 +4,8 @@
 
 namespace Shopsys\FrameworkBundle\Model\Product\Search\Export;
 
-class ProductSearchExportScheduler
-{
-    protected $productIds = [];
-
-    /**
-     * @param int $productId
-     */
-    public function scheduleProductIdForImmediateExport(int $productId): void
-    {
-        $this->productIds[] = $productId;
-    }
-
-    /**
-     * @return bool
-     */
-    public function hasAnyProductIdsForImmediateExport(): bool
-    {
-        return $this->productIds !== [];
-    }
+use Shopsys\FrameworkBundle\Component\Elasticsearch\AbstractExportScheduler;
 
-    /**
-     * @return int[]
-     */
-    public function getProductIdsForImmediateExport(): array
-    {
-        return array_unique($this->productIds);
-    }
-
-    public function cleanScheduleForImmediateExport(): void
-    {
-        $this->productIds = [];
-    }
+class ProductSearchExportScheduler extends AbstractExportScheduler
```

Comment:
What about to change the name (everywhere) to something without the 'Search' as ES is used for many more things now. What do you think?

Reply:
We can change it to `ExportScheduler` if you think it is reasonable. Also with others like:

- `ProductSearchExportCronModule` -> `ExportCronModule`
- `ProductSearchExportSubscriber` -> `ExportSubscriber`
- `ProductSearchExportWithFilterRepository` -> `ExportWithFilterRepository`

---
I do not like the names too but I dont think it is wise to change it just for a better looking and no added value.

Reply:
If we decide to rename it anyway I would suggest to change the namespace too. From `Shopsys\FrameworkBundle\Model\Product\Search\Export` to `Shopsys\FrameworkBundle\Model\Product\Elasticsearch`